diff --git a/meson.build b/meson.build
index 9a2c157..db9ebca 100644
--- a/meson.build
+++ b/meson.build
@@ -79,7 +79,6 @@ test_c_args = [
   '-Werror=shadow',
   '-Werror=strict-prototypes',
   '-Werror=trigraphs',
-  '-Werror=undef',
   '-Werror=write-strings',
   '-Wformat-nonliteral',
   ['-Werror=format-security', '-Werror=format=2' ],
@@ -101,7 +100,6 @@ test_c_args = [
   '-Wstrict-prototypes',
   '-Wswitch-default',
   '-Wswitch-enum',
-  '-Wundef',
   '-Wuninitialized',
   '-Wunused',
   '-fno-strict-aliasing',
@@ -124,7 +122,7 @@ hidden_visibility_args = []
 if get_option('default_library') != 'static'
   if host_machine.system() == 'windows'
     config_h.set('DLL_EXPORT', true)
-    if cc.get_id() == 'msvc'
+    if cc.get_id() == 'msvc' or cc.get_id() == 'clang-cl'
       config_h.set('_DZL_EXTERN', '__declspec(dllexport) extern')
     elif cc.has_argument('-fvisibility=hidden')
       config_h.set('_DZL_EXTERN', '__attribute__((visibility("default"))) __declspec(dllexport) extern')
diff --git a/src/graphing/dzl-cpu-model.c b/src/graphing/dzl-cpu-model.c
index e3af205..3e5b4a4 100644
--- a/src/graphing/dzl-cpu-model.c
+++ b/src/graphing/dzl-cpu-model.c
@@ -21,7 +21,9 @@
 #include <ctype.h>
 #include <fcntl.h>
 #include <stdlib.h>
-#include <unistd.h>
+#ifdef __linux__
+# include <unistd.h>
+#endif
 #include <stdio.h>
 #if defined(__FreeBSD__)
 # include <errno.h>
diff --git a/src/graphing/dzl-graph-line-renderer.c b/src/graphing/dzl-graph-line-renderer.c
index 3da778b..079b142 100644
--- a/src/graphing/dzl-graph-line-renderer.c
+++ b/src/graphing/dzl-graph-line-renderer.c
@@ -19,6 +19,7 @@
 #include "config.h"
 
 #include <glib/gi18n.h>
+#define _USE_MATH_DEFINES
 #include <math.h>
 
 #include "dzl-graph-line-renderer.h"
diff --git a/src/meson.build b/src/meson.build
index eeb20c5..7092a86 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -110,7 +110,6 @@ libdazzle = shared_library(
          dependencies: libdazzle_deps,
   include_directories: [ root_inc, src_inc ],
               install: true,
-          install_dir: libdir
 )
 
 libdazzle_dep = declare_dependency(
diff --git a/src/panel/dzl-tab-strip.c b/src/panel/dzl-tab-strip.c
index 5f892a3..50cfe49 100644
--- a/src/panel/dzl-tab-strip.c
+++ b/src/panel/dzl-tab-strip.c
@@ -771,7 +771,7 @@ apply_style (GtkWidget *widget,
     dzl_tab_set_style (DZL_TAB (widget), style);
 }
 
-guint
+DzlTabStyle
 dzl_tab_strip_get_style (DzlTabStrip *self)
 {
   DzlTabStripPrivate *priv = dzl_tab_strip_get_instance_private (self);
diff --git a/src/util/dzl-counter.c b/src/util/dzl-counter.c
index f57f936..f32eccb 100644
--- a/src/util/dzl-counter.c
+++ b/src/util/dzl-counter.c
@@ -30,7 +30,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/types.h>
+#ifndef G_OS_WIN32
 #include <unistd.h>
+#endif
 
 #ifdef G_OS_UNIX
 # include <sys/mman.h>
@@ -148,6 +150,7 @@ dzl_counter_reset (DzlCounter *counter)
   DZL_MEMORY_BARRIER;
 }
 
+#ifndef G_OS_WIN32
 static void
 _dzl_counter_arena_atexit (void)
 {
@@ -158,6 +161,7 @@ _dzl_counter_arena_atexit (void)
   g_snprintf (name, sizeof name, NAME_FORMAT, pid);
   shm_unlink (name);
 }
+#endif
 
 static void
 _dzl_counter_arena_init_local (DzlCounterArena *arena)
diff --git a/src/util/dzl-pango.c b/src/util/dzl-pango.c
index 05414c1..dfaa361 100644
--- a/src/util/dzl-pango.c
+++ b/src/util/dzl-pango.c
@@ -22,6 +22,7 @@
 
 #include <gdk/gdk.h>
 #include <glib/gstdio.h>
+#define _USE_MATH_DEFINES
 #include <math.h>
 
 #include "util/dzl-pango.h"
diff --git a/src/widgets/dzl-column-layout.c b/src/widgets/dzl-column-layout.c
index 37f1409..6ef47af 100644
--- a/src/widgets/dzl-column-layout.c
+++ b/src/widgets/dzl-column-layout.c
@@ -18,6 +18,7 @@
 
 #include "config.h"
 
+#define _USE_MATH_DEFINES
 #include <math.h>
 
 #define G_LOG_DOMAIN "dzl-column-layout"
diff --git a/src/widgets/dzl-progress-icon.c b/src/widgets/dzl-progress-icon.c
index 1baf488..7458f4b 100644
--- a/src/widgets/dzl-progress-icon.c
+++ b/src/widgets/dzl-progress-icon.c
@@ -21,6 +21,7 @@
 
 #include "config.h"
 
+#define _USE_MATH_DEFINES
 #include <math.h>
 
 #include "dzl-progress-icon.h"
diff --git a/src/widgets/dzl-three-grid.c b/src/widgets/dzl-three-grid.c
index d38eb8f..52507a5 100644
--- a/src/widgets/dzl-three-grid.c
+++ b/src/widgets/dzl-three-grid.c
@@ -20,6 +20,7 @@
 
 #include "config.h"
 
+#define _USE_MATH_DEFINES
 #include <math.h>
 
 #include "util/dzl-macros.h"

